### DISCUSSION TITLE
How to generate a final sql with prisma without having to

### DISCUSSION THREAD

**Discussion by dazitocode** [prisma/prisma#24875] [Q&A]
### Question

I want after developing, i can generate a final sql that will just be a single sql file without having to merge different migrations that prisma have generated at different stages

This way I can utilize the create table statements generated by prisma on other deployment mediums without having to run prisma again on that medium

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @dazitocode 👋 

To generate a single SQL file that consolidates all your migrations, you can use the `prisma migrate diff` command. This command allows you to compare two schemas and generate the SQL statements needed to evolve from one schema to the other.

Here is an example command to generate the SQL statements from an empty schema to your current schema:
```
npx prisma migrate diff \
  --from-empty \
  --to-schema-datamodel ./prisma/schema.prisma \
  --script > ./prisma/migrations/000000000000_squashed_migrations/migration.sql
```
This command uses the following options:

- --from-empty: The source for the SQL statement is an empty schema.
- --to-schema-datamodel ./prisma/schema.prisma: The target for the SQL statement is the data model in ./prisma/schema.prisma.
- --script: Output the result as SQL. If you omit this option, the "migration steps" will be generated in plain English.

After running this command, `./prisma/migrations/000000000000_squashed_migrations/migration.sql` will contain the SQL statements needed to create your entire schema from scratch. You can refer to the [Prisma documentation on squashing migrations](https://www.prisma.io/docs/orm/prisma-migrate/workflows/squashing-migrations).

This approach allows you to utilize the create table statements generated by Prisma on other deployment mediums without having to run Prisma again on that medium.

**Reply by zloom** [reply]
Worked for me, it was hard to use shell so i created tool on top of migrate package https://github.com/zloom/prisma-sql-gen

**Comment by CodeSmith32** [comment]
New versions of Prisma use `--to-schema` instead of `--to-schema-datamodel` and require `--config`:

```bash
npx prisma migrate diff \
  --config ./path/to/prisma.config.ts \
  --from-empty \
  --to-schema ./path/to/schema.prisma \
  --script > ./export/path/migration.sql
```

One line:

```bash
npx prisma migrate diff --config ./path/to/prisma.config.ts --from-empty --to-schema ./path/to/schema.prisma --script > ./export/path/migration.sql
```

(In Prisma 7+, `--config` is **required** even though it's not documented as such. See https://github.com/prisma/prisma/issues/28888)
